### ISSUE TITLE
Gradlew  Error (and Solution) When Running on Win 10

### ISSUE THREAD

**Issue by tajmone** [asciidoctor/asciidoctor-fopub#88]
After following the setup instructions, under Win 10, and having added ascidoctor-fopub to the system PATH, I kept getting an error about the installapp task not being available in my project folder (located elsewhere).

I've solved it by opening a CMD in the "asciddoctor-fopub" root folder and typing:

```
gradlew installapp
```

... after that, it all worked fine. So it might be worth mentioning this step in the README file, for it's not taken care of automatically nor it's mentioned.

I've also noticed at [line 18 of `fopub.bat`]:

```batch
set FOPUB_CMD=%FOPUB_DIR%\bin\fopub.bat
```

... but there is no `fopub.bat` file inside `bin`! only in the project's root.

It didn't seem to affect actual usage, but I thought it might be worth mentioning it.

[line 18 of `fopub.bat`]: https://github.com/asciidoctor/asciidoctor-fopub/blob/a5d8ee4/fopub.bat#L18

**Comment by mojavelinux** [comment]
I wonder if #94 fixes this issue.

**Comment by tajmone** [comment]
Will Gradle 4.9 also allow using Java 10? (see #87)

If I remember correctly, new versions of Gradle fixed the "time-based release model" bug that prevented using JDK 10. If this fixes the issue it would be really cool, as it would allow me to revert to Java 10 (had to step down to JDK 8 specifically to support asciidoctor-fopub).

When I find some time I'll try to setup from scratch the whole  asciidoctor-fopub I use, so I can give you some feedback on the previous issues (and if they're now fixed) and also try to use it with JDK 10.

**Comment by mojavelinux** [comment]
> Will Gradle 4.9 also allow using Java 10? 

That's precisely what I'm asking. I'll be interested to know.

**Comment by tajmone** [comment]
@mojavelinux,

finally today I've managed to find some time to update asciidoctor-fopub and Java.

It seems that the Gradle problem with JDK >= 10 persists. After having installed Java 11 on my system (Win 10 x64) I get the error:

```
D:\GitHub\AsciiDoc\asciidoctor-fopub>gradlew.bat

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine java version from '11.0.1'.
```

So it looks like the error mentioned ad #87 persists.

**Comment by tajmone** [comment]
As for the original problem of this post (the need to manually run `gradlew installapp`), it seems that the problem also persists.

After having reverted to Java 8, when trying to build a project I still get:

```
Initializing application...

FAILURE: Build failed with an exception.

* What went wrong:
Task 'installApp' not found in root project '_assets'.

* Try:
Run gradlew tasks to get a list of available tasks. Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
```

But if I manually run `gradlew installapp`, then everything works fine again:

```
D:\GitHub\AsciiDoc\asciidoctor-fopub>gradlew installapp
:extractDocbookXml
:extractDocbookXsl
:compileJava
:processResources UP-TO-DATE
:classes
:jar
:startScripts
:installApp
The installApp task has been deprecated and is scheduled to be removed in Gradle 3.0. Please use the installDist task instead.

BUILD SUCCESSFUL

Total time: 33.431 secs
```

There is a deprecation warning about using `installDist` instead, but it worked fine nonetheless.

**Comment by mojavelinux** [comment]
As for installApp vs installDist, that's a Gradle version thing. fopub now requires that you use Gradle 4 (which is what gradlew is bundling). If you use Gradle 4, you should not see that error.

**Comment by mojavelinux** [comment]
> finally today I've managed to find some time to update asciidoctor-fopub and Java.

I don't think you tested the Gradle 4.9 switch (because it was still in a PR). I've merged it, so now you can test again.

**Comment by tajmone** [comment]
## Works with Java JDK 12

HI @mojavelinux , sorry for the long delay in replying to this, but I just didn't find enough time to go through the whole process.

I now can confirm that I've manged to successfully use __asciidoctor-fopub__ with Java Java SE Development Kit 12.0.2 after pulling current master (and #94).

__NOTE__ — I don't quite remember why I had annotated to update via `gradlew installapp`, but the command failed with:

    FAILURE: Build failed with an exception.

    * What went wrong:
    Task 'installapp' not found in root project 'fopub'.

But running `gradlew tasks` finished the update and when I tried rebuilding a PDF ebook all went fine (unlike before).

Could it be that the  `gradlew installapp` command was mentioned in eaerlier versions of __asciidoctor-fopub__? Now I don't see any notes on how to finish the setup, is it handled automatically on first invocation?

**Comment by tajmone** [comment]
Ahh! The command is now `gradlew  installDist`!

The task succeeded, but with some warnings:

```
BUILD SUCCESSFUL in 1s
1 actionable task: 1 executed
D:\GitHub\AsciiDoc\asciidoctor-fopub>fopub.bat
Syntax error!
Usage: fopub [-t PATH] [-f FORMAT] [-H] FILE
Example: fopub -t \path\to\custom\docbook-xsl mydoc.xml


> Task :compileJava
POM relocation to an other version number is not fully supported in Gradle : xml-apis:xml-apis:2.0.2 relocated to xml-apis:xml-apis:1.0.b2.
Please update your dependency to directly use the correct version 'xml-apis:xml-apis:1.0.b2'.
Resolution will only pick dependencies of the relocated element.  Artifacts and other metadata will be ignored.
Note: D:\GitHub\AsciiDoc\asciidoctor-fopub\src\main\java\org\apache\fop\cli\InputHandler.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

BUILD SUCCESSFUL in 58s
6 actionable tasks: 6 executed
```

As you can see, it still complains about the missing `fopub.bat` mentioned at the beginning of this post.
